build_runner_core/lib/src/asset_graph/node.dart

```diff
@@ -273,24 +272,25 @@ class GlobAssetNode extends InternalAssetNode implements NodeWithInputs {
   final int phaseNumber;
 
   /// The actual results of the glob.
-  List<AssetId> results;
+  List<AssetId>? results;
 
   @override
   NodeState state;
 
   GlobAssetNode(AssetId id, this.glob, this.phaseNumber, this.state,
-      {this.inputs, Digest lastKnownDigest, this.results})
-      : super(id, lastKnownDigest: lastKnownDigest);
+      {HashSet<AssetId>? inputs, Digest? lastKnownDigest, this.results})
+      : inputs = inputs ?? HashSet(),
+        super(id, lastKnownDigest: lastKnownDigest);
 
   static AssetId createId(String package, Glob glob, int phaseNum) => AssetId(
       package, 'glob.$phaseNum.${base64.encode(utf8.encode(glob.pattern))}');
 }
 
 /// A node which has [inputs], a [NodeState], and a [phaseNumber].
 abstract class NodeWithInputs implements AssetNode {
-  HashSet<AssetId> inputs;
+  abstract HashSet<AssetId> inputs;
```

Comment:
Can this be a getter?

Reply:
It also needs a setter, at least the way the code is set up to use it today.

Reply:
done